### PR TITLE
S3168 FN: Mark virtual interface method test case as compliant.

### LINKED ISSUE #6305: Fix S3168 FN: Support virtual interface methods
### Description

FN for S3168 (AsyncVoidMethod).

Using C#11 virtual interface methods, the rule is not raising an issue.

### Repro steps

```csharp
public interface ISomeInterface
{
    static virtual async void SomeVirtualMethod() // FN
    {
        return;
    }
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Fixes #6305

### CODE REVIEW

**Comment by csaba-sagi-sonarsource** [comment]
Please rename the issue and leave a comment why it is compliant.
Based on the issue title one would think it fixes an FN however it actually marks the "FN" compliant.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
One comment was left about the titles of the PR and issue.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AsyncVoidMethod.CSharp11.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ public interface ISomeInterface
     {
         static abstract void SomeMethod1();
 
-        static virtual async void SomeVirtualMethod() // FN
+        static virtual async void SomeVirtualMethod() // Compliant (virtual member)
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AsyncVoidMethod.CSharp11.cs:9]: The rule explicitly ignore abstract/virtual methods as the implementation in the class might end up being non-async.

**Review by gregory-paidis-sonarsource — APPROVED** [review]

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**